logstash-core/lib/logstash/settings.rb

```diff
@@ -542,6 +543,51 @@ def validate(value)
       end
     end
 
+    class ValidatedPassword < Setting::Password
+      def initialize(name, value, password_policies)
+        @password_policies = password_policies
+        super(name, value, true)
+      end
+
+      def coerce(password)
+        if password && !password.kind_of?(::LogStash::Util::Password)
+          raise(ArgumentError, "Setting `#{name}` could not coerce LogStash::Util::Password value to password")
+        end
+
+        policies = set_password_policies
+        errors = LogStash::Util::PasswordValidator.new(policies).validate(password.value)
+        if errors.length() > 0
+          if @password_policies.fetch(:mode).eql?("WARN")
+            logger.warn("Password #{errors}.")
+            deprecation_logger.deprecated("NOTICE: 'WARN' password policy mode won't be allowed in the future. Set the mode to 'ERROR' to avoid startup errors in future releases.")
```

Comment:
@jsvd how do you think if we WARN list of unmet password validations and give deprecation notice?

Reply:
> I'm good with that. I could see an alternative where we we'd move the `logger.warn` to PasswordValidator so that users can mute the warning if they so wish by changing the logger level of this class to ERROR in log4j2.properties.

Yeah, one caveat I found with that way is, when Logstash raises an error (on BLOCK), same messages will be shown twice (in PasswordValidator.class and settings.rb 564L).